Tests/Set-ProGetAsset.Tests.ps1

```diff
@@ -20,52 +22,65 @@ function GivenSession
     }
 }
 
-function GivenAsset
+function GivenAssetName
 {
     param(
         [string]
-        $Name,
-        [string]
-        $RootDirectory,
-        [string]
-        $FilePath
+        $Name
     )
+
     $script:proGetAssetName = $Name
-    New-Item -Path (Join-Path -Path $TestDrive.FullName -ChildPath $FilePath) -ItemType 'File' -Force
-    if($RootDirectory)
-    {
-        $script:baseDirectory = $RootDirectory
-    }
-    $script:filePath = (Join-Path -Path $TestDrive.FullName -ChildPath $FilePath)
 }
 
-function GivenAssetWithoutFile
+function GivenAssetDirectory
 {
     param(
         [string]
-        $Name,
+        $RootDirectory        
+    )
+
+    $script:baseDirectory = $RootDirectory
+}
+
+function GivenSourceFilePath
+{
+    param(
         [string]
-        $FilePath
+        $Path,
+        [switch]
+        $WhereFileDoesNotExist
     )
-    $script:proGetAssetName = $Name
-    $script:filePath = $FilePath
 
+    $script:filePath = Join-Path -Path $TestDrive.FullName -ChildPath $Path
+
+    if( !$WhereFileDoesNotExist )
+    {
+        New-Item -Path $filePath -ItemType 'File' -Force
+    }
 }
 
-function GivenAssetThatDoesntExist
+function GivenSourceBodyContent
 {
     param(
         [string]
-        $Name
+        $Body
     )
-    $script:proGetAssetName = $Name
-    $script:proGetAssetDirectory = $null
+
+    $script:bodyContent = $Body
 }
 
-function WhenAssetIsUploaded
+function WhenAssetIsPublished
 {
     $Global:Error.Clear()
-    Set-ProGetAsset -Session $session -Path $proGetAssetName -DirectoryName $baseDirectory -FilePath $filePath -ErrorAction SilentlyContinue
+    
+    if( $filePath )

```

Comment:
Use a splatted hashtable instead of calling `Set-ProGetAsset` twice, e.g.

```
$parameter = @{ }
if( $filePath )
{
     $parameter['FilePath'] = $filePath
}
else
{
   $parameter['Body'] = $Value
}
Set-ProGetAsset -Session $session -Path $proGetAssetName -DirectoryName $baseDirectory @parameter
```